### PR TITLE
Upgrade to SpringBoot3.0

### PR DESCRIPTION
**Pull request by ziyanli-amazon**
*Issue #, if available:*
Upgraded to spring boot3
*Description of changes:*
- Changed signature to match `jakarta.jms`
- Upgraded to mickto5 and junit5
- Lintered code 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by AlexRosier** [comment]
@greensopinion and or @eddy-aws Could you guys take a look at this ? It is currently blocking my team and I assume many other from going to spring boot 3

**Comment by okohub** [comment]
up up up

File: src/main/java/com/amazon/sqs/javamessaging/util/ExponentialBackoffStrategy.java [hunk 1]
```diff
@@ -33,7 +33,7 @@ public ExponentialBackoffStrategy(long delayInterval, long initialDelay, long ma
     /**
      * Returns the delay before the next attempt.
      * 
-     * @param retriesAttempted
+     * @param retriesAttempted number of retries attempted
```
Comment (eddy-aws) [inline src/main/java/com/amazon/sqs/javamessaging/util/ExponentialBackoffStrategy.java:36]: This parameter description doesn't really convey more than what the variable conveys. In such cases, I think we should just skip the description.

File: src/test/java/com/amazon/sqs/javamessaging/SQSSessionTest.java [hunk 2]
```diff
@@ -1482,19 +1284,14 @@ public void testCreateObjectMessage() throws JMSException, InterruptedException
...
          */
-        try {
-            sqsSession.createTextMessage();
-            fail();
-        } catch(IllegalStateException ise) {
-            assertEquals("Session is closed", ise.getMessage());
-        }
+        assertThrows(IllegalStateException.class, () -> sqsSession.createTextMessage(), "Session is closed");
 
         sqsSession.setClosed(false);
```
Comment (eddy-aws) [inline src/test/java/com/amazon/sqs/javamessaging/SQSSessionTest.java:1296]: Wow, that was a huge test, may be we should consider splitting the file in the future for clarity.

**Review by eddy-aws — APPROVED** [review]
Left some minor comments/questions. But approving to merge this change. Do we have integration tests for this package?

File: src/main/java/com/amazon/sqs/javamessaging/SQSConnection.java [hunk 3]
```diff
@@ -410,7 +418,7 @@ public void close() throws JMSException {
 
                 }
             }
-        }/** Blocks until closing of the connection completes */
+        }// Blocks until closing of the connection completes
```
[the remark this answers is not in the document]
Comment (ziyanli-amazon) [reply src/main/java/com/amazon/sqs/javamessaging/SQSConnection.java:421]: Agree. Will remove comments like this in the next PR. The change was made due to the compliance of javadoc warning.

File: src/main/java/com/amazon/sqs/javamessaging/util/ExponentialBackoffStrategy.java (hunk 1 above)
Comment (ziyanli-amazon) [reply src/main/java/com/amazon/sqs/javamessaging/util/ExponentialBackoffStrategy.java:36]: agree. will change it

File: src/test/java/com/amazon/sqs/javamessaging/SQSSessionTest.java (hunk 2 above)
Comment (ziyanli-amazon) [reply src/test/java/com/amazon/sqs/javamessaging/SQSSessionTest.java:1296]: Will do. Need to do an assessment first. thanks for the suggestion.

**Pull request merged**